### PR TITLE
[PM-22202] Perform sync in Fido2 findCredentials when they have counter

### PR DESCRIPTION
**Pull request by fedemkr**
## 🎟️ Tracking

[PM-22202](https://bitwarden.atlassian.net/browse/PM-22202)

## 📔 Objective

Performance improvement to only sync credentials in Fido2 flows when there are Fido2 credentials with `counter > 0` and the sync is needed (30 min have passed since last one).

> [!IMPORTANT]  
> `findCredentials` now calls a private function to prevent infinite loop while using recursion in some edge cases sync may present.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-22202]: https://bitwarden.atlassian.net/browse/PM-22202?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/Core/Vault/Services/Fido2CredentialStoreService.swift [hunk 1]
```diff
@@ -55,18 +61,46 @@ final class Fido2CredentialStoreService: Fido2CredentialStore {
...
+    // MARK: Private methods
 
+    /// Finds active login ciphers that have Fido2 credentials, match the `ripId` and if `ids` is sent
+    /// then filters the one which the Fido2 `credentialId` matches some of the one in `ids`.
+    /// - Parameters:
+    ///   - ids: An array of possible `credentialId` to filter credentials that matches one of them.
+    ///   When `nil` the `credentialId` filter is not applied.
+    ///   - ripId: The `ripId` to match the Fido2 credential `rpId`.
+    ///   - shouldCheckSync: Whether it should check if sync is needed. This is particular useful to vaoid
+    ///   inifinite loops by calling this method recursively.
```
Comment (matt-livefront) [inline BitwardenShared/Core/Vault/Services/Fido2CredentialStoreService.swift]: ⛏️ 
```suggestion
    ///   - shouldCheckSync: Whether it should check if sync is needed. This is particular useful to avoid
    ///   infinite loops by calling this method recursively.
```
